pandas/tests/frame/methods/test_to_dict.py

```diff
@@ -513,6 +513,18 @@ def test_to_dict_masked_native_python(self):
         result = df.to_dict(orient="records")
         assert isinstance(result[0]["a"], int)
 
+    def test_to_dict_tight_no_warning_with_duplicate_column(self):
+        # GH#58281
+        df = DataFrame(
+            {
+                "A": [1, 2, 3],
+                "B": [4, 5, 6],
+            }
+        )
+        df.columns = ["A", "A"]
```

Comment:
I would recommend setting the columns attribute via the DataFrame constructor I.E. 
```python
df = DataFrame([[1, 2], [3, 4], [5, 6]], columns=["A", "A"])
```